.github/workflows/ci.yml

```
@@ -51,33 +51,23 @@ jobs:
           # Don't use a 'components:' entry--we don't need them with beta/nightly, plus nightly often doesn't have them
           override: true
 
-      - name: install rustfmt+clippy on stable
-        run: |
-          rustup component add rustfmt
-          rustup component add clippy
+      - run: rustup component add rustfmt clippy
         if: contains(matrix.toolchain, 'stable')
 
-      - name: run rustfmt on stable
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+      - run: cargo fmt
         if: contains(matrix.toolchain, 'stable')
 
-      - name: run clippy on stable
-        uses: actions-rs/cargo@v1
-        with:
-          command: clippy
-          args: --features=${{matrix.FEATURES}}
+      - run: cargo clippy --features=${{matrix.FEATURES}}
```

Aha! You've discovered another bug. We were only running clippy on the root crate.  We can't make it fail yet, though, because we haven't taken care of the warnings.
```suggestion
      - run: cargo clippy --features=${{matrix.FEATURES}} --workspace
```

---

Yup, added `--workspace` in 6fcda3a

---

Surprisingly, with clippy hasn't detected any issues for the rest of the codebase. I guess that `Ran cargo clippy` chebox in PR template did the thing?

---

I won't complain if the clippy issues are all fixed! If that is the case, let's make clippy fail if it finds anything.

---

@CleanCut 
Well actually clippy does detect issues, but only at warning level and it doesn't fail ci, see [this line in checks log](https://github.com/amethyst/amethyst/pull/2443/checks?check_run_id=1006732344#step:7:197)

Would you merge this if I add these to to global ci env vars we have in our project?
```
  RUSTFLAGS: "--deny warnings -Cdebuginfo=0" // Disabling debug info should speed up ci?
  RUSTDOCFLAGS: "--deny warnings"
```